avocado/utils/linux_modules.py

```diff
@@ -208,34 +208,107 @@ def get_loaded_modules():
         return [astring.to_text(_.split(b' ', 1)[0]) for _ in proc_modules]
 
 
-def check_kernel_config(config_name):
+def check_kernel_config(config_name, session=None):
     """
     Reports the configuration of $config_name of the current kernel
 
     :param config_name: Name of kernel config to search
+    :param session: guest session, command is run on host if None
     :type config_name: str
     :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
     :rtype: :class:`ModuleConfig`
     """
 
-    kernel_version = platform.uname()[2]
+    def check_local_kernel_config(config_name):
+        """
+        Reports the configuration of $config_name of the current kernel
+
+        :param config_name: Name of kernel config to search
+        :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
+        :rtype: :class:`ModuleConfig`
+        """
+        kernel_version = platform.uname()[2]
+
+        config_file = '/boot/config-' + kernel_version
+        with open(config_file, 'r') as kernel_config:  # pylint: disable=W1514
+            for line in kernel_config:
+                line = line.split('=')
+
+                if len(line) != 2:
+                    continue
+                config = line[0].strip()
+                if config == config_name:
+                    option = line[1].strip()
+                    if option == "m":
+                        return ModuleConfig.MODULE
+                    else:
+                        return ModuleConfig.BUILTIN
+        return ModuleConfig.NOT_SET
+
+    def check_session_kernel_config(config_name, session):
+        """
+        Reports the configuration of $config_name of session's kernel
+
+        :param config_name: Name of kernel config to search
+        :param session: remote session
+        :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
+        :rtype: :class:`ModuleConfig`
+        """
+        kernel_version = session.cmd('uname -r').strip()
```

Comment:
> This should be:
> 
> ```python
> kernel_version = session.cmd('uname -r').stdout_text.strip()
> ```
> 
> Because `avocado.utils.ssh.Session.cmd()` returns a `CmdResult` instance.

Hi @clebergnu 
I just realized a serious problem:
The `session` I  used here is not a avocado.utils.ssh.Session, is an aexpect.ShellSession...
aexpect.ShellSession reutrns string directly. That is why I didn't meet any error during test... 
```
[stdlog] 2022-07-04 04:02:13,241 avocado.test ERROR|   File "/root/avocado-build/avocado/avocado/utils/linux_modules.py", line 277, in check_kernel_config
[stdlog] 2022-07-04 04:02:13,241 avocado.test ERROR|     check_session_kernel_config(config_name, session)
[stdlog] 2022-07-04 04:02:13,241 avocado.test ERROR|   File "/root/avocado-build/avocado/avocado/utils/linux_modules.py", line 256, in check_session_kernel_config
[stdlog] 2022-07-04 04:02:13,241 avocado.test ERROR|     kernel_version = session.cmd('uname -r').stdout_text.strip()
[stdlog] 2022-07-04 04:02:13,242 avocado.test ERROR| AttributeError: 'str' object has no attribute 'stdout_text'
```

I submit this commit b/c I have a tp-libvirt case which depends on it. 
But the session of tp-libvirt was from avocado-vt (aexpect shellsession),
not the avocado.utils.ssh.Session...

I will complete this PR for avocado.utils.ssh.Session, then open a new PR in avocado-vt.